pkg/op/server_http.go

```diff
@@ -29,15 +29,18 @@ func RegisterServer(server Server, endpoints Endpoints, options ...ServerOption)
 		server:    server,
 		endpoints: endpoints,
 		decoder:   decoder,
+		corsOpts:  &defaultCORSOptions,
 		logger:    slog.Default(),
 	}
-	ws.router.Use(cors.New(defaultCORSOptions).Handler)
 
 	for _, option := range options {
 		option(ws)
 	}
 
 	ws.createRouter()
+	if ws.corsOpts != nil {
+		return cors.New(*ws.corsOpts).Handler(ws)
+	}
```

Comment:
Tests currently panic on this, as the test does type assertion to check the server state.  I tried moving the function registration to the router, only then chi panics because it wants middleware before defining routes. And currently the `LegacyServer`  defines a custom route as part of ServerOptions.

The only way I see now to add CORS to the Server is to use the `WithHTTPMiddleware` directly. I don't have time now to get my head around it, so I'll get back to it later.